pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterUpdater.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.minion;
+
+import org.apache.pinot.common.minion.MinionTaskMetadataUtils;
+import org.apache.pinot.common.minion.RealtimeToOfflineSegmentsTaskMetadata;
+import org.apache.pinot.controller.helix.core.PinotHelixResourceManager;
+import org.apache.pinot.controller.helix.core.minion.generator.PinotTaskGenerator;
+import org.apache.pinot.core.common.MinionConstants;
+
+
+/**
+ * An abstraction on top of {@link PinotHelixResourceManager}, created for the {@link PinotTaskGenerator},
+ * with scope restricted to cluster updates.
+ * This also helps in separating read and update methods from the {@link ClusterInfoProvider}
+ */
+public class ClusterUpdater {
```

Comment:
After some reconsideration, I feel it might be easier to manage if we merge provider and updater to `ClusterInfoAccessor` in case we need update the metadata based on the current value in the future.

Sorry that I was suggesting separating it out in the previous review.